### ISSUE TITLE
for loop with range inside nested class assignment

### ISSUE THREAD

**Issue by bouk** [juliankrispel/decaf#64]
This throws an error for me on master:

```
class A.B
  c: ->
    for d in [0..e]
      d

```

It works fine if it's not a nested class (so `class A` instead)

trace:

```
Skipping file "input.coffee" because of error.
TypeError: Cannot read property 'freeVariable' of undefined
  at Range.exports.Range.Range.compileArray (/Users/bouke/repos/decaf/node_modules/coffee-script/lib/coffee-script/nodes.js:1209:18)
  at Range.exports.Range.Range.compileNode (/Users/bouke/repos/decaf/node_modules/coffee-script/lib/coffee-script/nodes.js:1170:21)
  at Range.exports.Base.Base.compileToFragments (/Users/bouke/repos/decaf/node_modules/coffee-script/lib/coffee-script/nodes.js:83:21)
  at Value.exports.Value.Value.compileNode (/Users/bouke/repos/decaf/node_modules/coffee-script/lib/coffee-script/nodes.js:812:29)
  at Value.exports.Base.Base.compileToFragments (/Users/bouke/repos/decaf/node_modules/coffee-script/lib/coffee-script/nodes.js:83:21)
  at Value.exports.Base.Base.compile (/Users/bouke/repos/decaf/node_modules/coffee-script/lib/coffee-script/nodes.js:71:35)
  at mapRange (/Users/bouke/repos/decaf/dist/index.js:155:103)
  at mapValue (/Users/bouke/repos/decaf/dist/index.js:169:12)
  at mapExpression (/Users/bouke/repos/decaf/dist/index.js:1078:12)
  at mapForStatement (/Users/bouke/repos/decaf/dist/index.js:920:202)
  at mapStatement (/Users/bouke/repos/decaf/dist/index.js:510:12)
  at /Users/bouke/repos/decaf/dist/index.js:526:12
  at Array.map (native)
  at mapBlockStatements (/Users/bouke/repos/decaf/dist/index.js:525:27)
  at mapBlockStatement (/Users/bouke/repos/decaf/dist/index.js:533:23)
  at mapFunction (/Users/bouke/repos/decaf/dist/index.js:744:41)
  at mapExpression (/Users/bouke/repos/decaf/dist/index.js:1068:12)
  at mapClassBodyElement (/Users/bouke/repos/decaf/dist/index.js:279:96)
  at /Users/bouke/repos/decaf/dist/index.js:319:16
  at Array.map (native)
  at /Users/bouke/repos/decaf/dist/index.js:318:37
  at Array.reduce (native)
  at mapClassExpressions (/Users/bouke/repos/decaf/dist/index.js:308:22)
  at mapClassBody (/Users/bouke/repos/decaf/dist/index.js:331:23)
  at mapClassExpression (/Users/bouke/repos/decaf/dist/index.js:363:81)
  at mapClassDeclaration (/Users/bouke/repos/decaf/dist/index.js:370:132)
  at mapStatement (/Users/bouke/repos/decaf/dist/index.js:512:12)
  at /Users/bouke/repos/decaf/dist/index.js:526:12
  at Array.map (native)
  at mapBlockStatements (/Users/bouke/repos/decaf/dist/index.js:525:27)
  at mapBlockStatement (/Users/bouke/repos/decaf/dist/index.js:533:23)
  at transpile (/Users/bouke/repos/decaf/dist/index.js:1187:17)
  at /Users/bouke/repos/decaf/dist/index.js:1202:12
  at /Users/bouke/repos/decaf/node_modules/lodash/internal/createFlow.js:67:31
  at Object.compile (/Users/bouke/repos/decaf/dist/index.js:1205:10)
  at runByFile (/Users/bouke/repos/decaf/bin/decaf:52:20)
  at /Users/bouke/repos/decaf/bin/decaf:44:7
  at Array.forEach (native)
  at run (/Users/bouke/repos/decaf/bin/decaf:28:9)
  at Object.<anonymous> (/Users/bouke/repos/decaf/bin/decaf:25:1)
  at Module._compile (module.js:413:34)
  at Object.Module._extensions..js (module.js:422:10)
  at Module.load (module.js:357:32)
  at Function.Module._load (module.js:314:12)
  at Function.Module.runMain (module.js:447:10)
  at startup (node.js:139:18)
  at node.js:999:3


```

**Comment by bouk** [comment]
Almost every bug I run in to seems to have something to do with the `class A.B` pattern that's common in the codebase I'm working on

**Comment by juliankrispel** [comment]
@bouk thanks I'll have a look shortly

**Comment by juliankrispel** [comment]
fixed in #69
